internal/konnect/client.go

```diff
@@ -0,0 +1,193 @@
+package konnect
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/adminapi"
+	tlsutil "github.com/kong/kubernetes-ingress-controller/v2/internal/util/tls"
+)
+
+// AdminClient is used for sending requests to Konnect APIs which are not included
+// in Kong Admin APIs, like node registration APIs or runtime group operation APIs.
+// TODO(naming): give a better type name to this client?
+type AdminClient struct {
```

Comment:
`Client` sounds ok to me. It's in the `konnect` package so it would be referenced as `konnect.Client`.

Unless there's an intention to introduce another client for konnect 🤔

Reply:
I prefer to use `Client` in case we will wrap other APIs in this client. If we want to use more specific clients (for example a client that only do CRUD of nodes in node agent), use some interfaces containing only the methods used.